// @snapshot-labs/sx/dist/client/types.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.voteTypes = exports.proposalTypes = exports.spaceTypes = exports.domain = void 0;
exports.domain = {
    name: 'snapshot-x',
    version: '1'
    // chainId: 'starknet', // Expect a type "uint256" here, following EIP-155
    // verifyingContract: '0x335c0d227fbbf8d11e8d7b3788a7547495acc5ec3bf9e5031e72915a900b8c5' // StarkNet contract where to send message (voting contract), expect type "address"
};
exports.spaceTypes = {
    Space: [
        { name: 'strategies', type: 'bytes32[]' },
        { name: 'votingDelay', type: 'uint32' },
        { name: 'votingPeriod', type: 'uint32' },
        { name: 'proposalThreshold', type: 'uint32' } // Minimum voting power to have to be allowed to post a proposal
    ]
};
exports.proposalTypes = {
    Proposal: [
        { name: 'executionHash', type: 'bytes32' },
        { name: 'metadataHash', type: 'bytes32' } // Hash of EIP-712 message that contain: title, body
    ]
};
exports.voteTypes = {
    Vote: [
        { name: 'proposal', type: 'bytes32' },
        { name: 'choice', type: 'uint32' } // Possible choice: 0 = against, 1 = for, 2 = abstain
    ]
};
